src/target/riscv/riscv.c

```diff
@@ -4107,32 +4125,42 @@ static int parse_ranges(struct list_head *ranges, const char *tcl_arg, const cha
 
 			if (entry->name && name && (strcasecmp(entry->name, name) == 0)) {
 				LOG_ERROR("Duplicate register name \"%s\" found.", name);
-				free(name);
-				free(args);
 				return ERROR_FAIL;
 			}
 		}
 
 		range_list_t *range = calloc(1, sizeof(range_list_t));
 		if (!range) {
 			LOG_ERROR("Failed to allocate range list.");
-			free(name);
-			free(args);
 			return ERROR_FAIL;
 		}
 
 		range->low = low;
 		range->high = high;
 		range->name = name;
+		/* ownership over name_buffer contents is transferred to list item here */
+		*name_buffer = NULL;
 		list_add(&range->list, ranges);
-
-		arg = strtok(NULL, ",");
 	}
 
-	free(args);
 	return ERROR_OK;
 }
 
+static int parse_reg_ranges(struct list_head *ranges, const char *tcl_arg,
+		const char *reg_type, unsigned int max_val)
+{
+	char *args = strdup(tcl_arg);
+	if (!args) {
+		LOG_ERROR("Failed to allocate memory for reg range string copy");
```

Comment:
Cosmetic detail: 

I'd simplify the error message so that the readers are not bothered with implementation details and instead get clear idea of what is going on:

```suggestion
		LOG_ERROR("Out of memory");
```

Or if you like:

```suggestion
		LOG_ERROR("Out of memory in " __func__);
```

Reply:
addressed